craft_application/services/provider.py

```diff
@@ -51,16 +52,18 @@ class ProviderService(base.BaseService):
 
     managed_mode_env_var = "CRAFT_MANAGED_MODE"
 
-    def __init__(
+    def __init__(  # noqa: PLR0913 (too many arguments)
```

Comment:
We do this enough that I've raised the question of whether maxing out at 5 arguments is a reasonable value for us. https://github.com/canonical/starbase/issues/133